### ISSUE TITLE
running postinstall script inside the app throws error

### ISSUE THREAD

**Issue by ishandutta2007** [chentsulin/electron-react-boilerplate#1612]
```
cd app
npm run electron-rebuild
```

```
> 
> 
> An unhandled error occurred inside electron-rebuild
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/surface/init.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/avl/avl.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/backoff/backoff.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/channel_args.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/channel_stack.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/channel_stack_builder.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/connected_channel.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/handshaker.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/handshaker_factory.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/handshaker_registry.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/compression.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/compression_internal.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/message_compress.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/stream_compression.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/stream_compression_gzip.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/stream_compression_identity.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/debug/stats.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/debug/stats_data.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/http/format_request.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/http/httpcli.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/http/parser.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/call_combiner.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/combiner.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/endpoint.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/endpoint_pair_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/endpoint_pair_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/endpoint_pair_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/error.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_epoll1_linux.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_epollex_linux.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_epollsig_linux.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_poll_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/exec_ctx.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/executor.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/fork_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/fork_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/gethostname_fallback.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/gethostname_host_name_max.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/gethostname_sysconf.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/iocp_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/iomgr.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/iomgr_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/iomgr_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/iomgr_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/is_epollexclusive_available.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/load_file.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/lockfree_event.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/network_status_tracker.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/polling_entity.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/pollset_set_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/pollset_set_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/pollset_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/pollset_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/resolve_address_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/resolve_address_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/resolve_address_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/resource_quota.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/sockaddr_utils.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_factory_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_mutator.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_utils_common_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_utils_linux.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_utils_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_utils_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_utils_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_client_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_client_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_client_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_utils_posix_common.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_utils_posix_ifaddrs.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_utils_posix_noifaddrs.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/time_averaged_stats.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/timer_generic.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/timer_heap.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/timer_manager.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/timer_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/udp_server.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/unix_sockets_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/unix_sockets_posix_noop.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/wakeup_fd_cv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/wakeup_fd_eventfd.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/wakeup_fd_nospecial.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/wakeup_fd_pipe.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/wakeup_fd_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/json/json.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/json/json_reader.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/json/json_string.o
>   CXX(target) child_process.js:644
>     throw err;
>     ^
> 
> Error: Command failed: ../node_modules/.bin/electron-rebuild --parallel --force --types prod,dev,optional --module-dir .
> 
> An unhandled error occurred inside electron-rebuild
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/surface/init.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/avl/avl.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/backoff/backoff.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/channel_args.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/channel_stack.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/channel_stack_builder.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/connected_channel.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/handshaker.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/handshaker_factory.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/channel/handshaker_registry.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/compression.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/compression_internal.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/message_compress.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/stream_compression.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/stream_compression_gzip.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/compression/stream_compression_identity.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/debug/stats.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/debug/stats_data.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/http/format_request.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/http/httpcli.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/http/parser.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/call_combiner.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/combiner.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/endpoint.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/endpoint_pair_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/endpoint_pair_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/endpoint_pair_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/error.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_epoll1_linux.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_epollex_linux.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_epollsig_linux.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_poll_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/ev_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/exec_ctx.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/executor.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/fork_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/fork_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/gethostname_fallback.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/gethostname_host_name_max.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/gethostname_sysconf.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/iocp_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/iomgr.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/iomgr_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/iomgr_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/iomgr_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/is_epollexclusive_available.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/load_file.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/lockfree_event.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/network_status_tracker.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/polling_entity.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/pollset_set_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/pollset_set_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/pollset_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/pollset_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/resolve_address_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/resolve_address_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/resolve_address_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/resource_quota.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/sockaddr_utils.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_factory_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_mutator.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_utils_common_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_utils_linux.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_utils_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_utils_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_utils_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/socket_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_client_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_client_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_client_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_utils_posix_common.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_utils_posix_ifaddrs.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_utils_posix_noifaddrs.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_server_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/tcp_windows.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/time_averaged_stats.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/timer_generic.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/timer_heap.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/timer_manager.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/timer_uv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/udp_server.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/unix_sockets_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/unix_sockets_posix_noop.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/wakeup_fd_cv.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/wakeup_fd_eventfd.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/wakeup_fd_nospecial.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/wakeup_fd_pipe.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/iomgr/wakeup_fd_posix.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/json/json.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/json/json_reader.o
>   CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/json/json_string.o
>   CXX(target) 
>     at checkExecSyncError (child_process.js:601:13)
>     at execSync (child_process.js:641:13)
>     at Object.<anonymous> (/Users/ishandutta2007/Documents/Projects/electron-react-boilerplate/internals/scripts/ElectronRebuild.js:21:3)
>     at Module._compile (module.js:635:30)
>     at loader (/Users/ishandutta2007/Documents/Projects/electron-react-boilerplate/node_modules/babel-register/lib/node.js:144:5)
>     at Object.require.extensions.(anonymous function) [as .js] (/Users/ishandutta2007/Documents/Projects/electron-react-boilerplate/node_modules/babel-register/lib/node.js:154:7)
>     at Module.load (module.js:554:32)
>     at tryModuleLoad (module.js:497:12)
>     at Function.Module._load (module.js:489:3)
>     at Function.Module.runMain (module.js:676:10)
```

**Comment by ishandutta2007** [comment]
interest using npm gives no error.

`ishandutta2007@MacBook-Pro:~/Documents/Projects/electron-react-boilerplate/app$ npm i --save firebase@5.0.1`

No error


`ishandutta2007@MacBook-Pro:~/Documents/Projects/electron-react-boilerplate/app$ yarn add firebase@5.0.1`
error

**Comment by amilajack** [comment]
It seems like one of your dependencies is causing you issues (`grpc`). See https://github.com/chentsulin/electron-react-boilerplate/issues/1601

**Comment by ishandutta2007** [comment]
ok but your code is not working.

**Comment by amilajack** [comment]
The code in the example? Can you show me what you get in the console

**Comment by ishandutta2007** [comment]
To start with I created this #1614 
Secondly even when I fixed it with something like this
```
import { dependencies }  from '../../app/package.json';
const modules = Object.keys(dependencies || {}).filter(e => e === 'grpc').join(',') || "";
```
to finally get `{$modules}` as a concatinated string of packanames like this "pkgA, pkgB, pkgC", I get


An unhandled error occurred inside electron-rebuild
argv.w.split is not a function

> 
> TypeError: argv.w.split is not a function
>     at Object.<anonymous> (/Users/ishandutta2007/Documents/Projects/electron-react-boilerplate/node_modules/electron-rebuild/lib/src/cli.js:118:39)
>     at Generator.next (<anonymous>)
>     at /Users/ishandutta2007/Documents/Projects/electron-react-boilerplate/node_modules/electron-rebuild/lib/src/cli.js:8:71
>     at new Promise (<anonymous>)
>     at __awaiter (/Users/ishandutta2007/Documents/Projects/electron-react-boilerplate/node_modules/electron-rebuild/lib/src/cli.js:4:12)
>     at __dirname (/Users/ishandutta2007/Documents/Projects/electron-react-boilerplate/node_modules/electron-rebuild/lib/src/cli.js:69:8)
>     at Object.<anonymous> (/Users/ishandutta2007/Documents/Projects/electron-react-boilerplate/node_modules/electron-rebuild/lib/src/cli.js:145:4)
>     at Module._compile (module.js:635:30)
>     at Object.Module._extensions..js (module.js:646:10)
>     at Module.load (module.js:554:32)
>

**Comment by ishandutta2007** [comment]
I am still struggling just trying to add these two lines to package.json, it might be very trivial to people handling this repo day in day out, would appriciate if someone can just do that and send me a fork.
```
    "react-firebaseui": "^3.0.4",
    "firebase": "^5.0.1"
```
The former is non-native, hence will go into top level and the later is native, hence should go to app folder level.

**Comment by Slapbox** [comment]
@ishandutta2007 in `./app`, try running `npm rebuild` and if that doesn't work try `npm rebuild --runtime=electron --target=2.0.2 --disturl=https://atom.io/download/atom-shell`

**Comment by Slapbox** [comment]
By the way, this is what my package.json had for dependencies there. I added `grpc` at the same version firebase was requiring. I don't know if this matters.

```
    "firebase": "5.1.0",
    "grpc": "1.11.3",
```

**Comment by ishandutta2007** [comment]
@Slapbox That part is working fine provided I have also changed the same in postinstall at bottom level but when I come back to top level and add `"react-firebaseui": "^3.0.4"` and yarn, now `"build-dll"` is causing issues. What should be postinstall at top level ?

> 
> $> electron-react-boilerplate@0.13.3 build-dll /Users/ishandutta2007/Documents/Projects/electron-react-boilerplate
> $> cross-env NODE_ENV=development node --trace-warnings -r babel-register ./node_modules/webpack/bin/webpack --config webpack.config.renderer.dev.dll.js --colors
> 
> Hash: fa43f17def65efdca84f
> Version: webpack 4.8.3
> Time: 5045ms
> Built at: 07/01/2018 6:15:22 AM
>               Asset      Size    Chunks             Chunk Names
> renderer.dev.dll.js  1.59 MiB  renderer  [emitted]  renderer
> Entrypoint renderer = renderer.dev.dll.js
> [./node_modules/electron-debug sync recursive] ./node_modules/electron-debug sync 160 bytes {renderer} [built]
> [./node_modules/webpack/buildin/amd-define.js] (webpack)/buildin/amd-define.js 88 bytes {renderer} [built]
> [./node_modules/webpack/buildin/harmony-module.js] (webpack)/buildin/harmony-module.js 597 bytes {renderer} [built]
> [electron] external "electron" 42 bytes {renderer} [built]
> [fs] external "fs" 42 bytes {renderer} [built]
> [module] external "module" 42 bytes {renderer} [built]
> [path] external "path" 42 bytes {renderer} [built]
> [0] dll renderer 12 bytes {renderer} [built]
>     + 139 hidden modules
> 
> WARNING in ./node_modules/electron-debug/index.js
> 81:45-58 Critical dependency: the request of a dependency is an expression
>  @ ./node_modules/electron-debug/index.js
>  @ dll renderer
> 
> WARNING in ./node_modules/electron-debug/index.js
> 84:85-106 Critical dependency: the request of a dependency is an expression
>  @ ./node_modules/electron-debug/index.js
>  @ dll renderer
> 
> ERROR in ./node_modules/firebaseui/dist/npm.js
> Module not found: Error: Can't resolve 'firebase/app' in '/Users/ishandutta2007/Documents/Projects/electron-react-boilerplate/node_modules/firebaseui/dist'
>  @ ./node_modules/firebaseui/dist/npm.js 1:27-50
>  @ ./node_modules/react-firebaseui/StyledFirebaseAuth.js
>  @ ./node_modules/react-firebaseui/index.js
>  @ dll renderer
> 
> ERROR in ./node_modules/firebaseui/dist/npm.js
> Module not found: Error: Can't resolve 'firebase/auth' in '/Users/ishandutta2007/Documents/Projects/electron-react-boilerplate/node_modules/firebaseui/dist'
>  @ ./node_modules/firebaseui/dist/npm.js 1:51-75
>  @ ./node_modules/react-firebaseui/StyledFirebaseAuth.js
>  @ ./node_modules/react-firebaseui/index.js
>  @ dll renderer
> npm ERR! code ELIFECYCLE
> npm ERR! errno 2
> npm ERR! electron-react-boilerplate@0.13.3 build-dll: `cross-env NODE_ENV=development node --trace-warnings -r babel-register ./node_modules/webpack/bin/webpack --config webpack.config.renderer.dev.dll.js --colors`
> npm ERR! Exit status 2
> npm ERR! 
> npm ERR! Failed at the electron-react-boilerplate@0.13.3 build-dll script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /Users/ishandutta2007/.npm/_logs/2018-07-01T00_45_22_663Z-debug.log
> error Command failed with exit code 2.
>

**Comment by ishandutta2007** [comment]
Solved by moving `"react-firebaseui": "^3.0.4"` to  lower level.
I think the doc [here](https://github.com/chentsulin/electron-react-boilerplate#which-packagejson-file-to-use) is not very clear 
instead of 
> all modules go into ./package.json except native modules. Native modules go into ./app/package.json

it should say 

Any native and non native which has atleast one native peer should go into `./app/package.json` and rest all into `./package.json` is what it should say.

**Comment by ishandutta2007** [comment]
I think I can close this one as long as #1601 is kep open, run `npm rebuild` whenever `Electron-rebuild` fails doesn't seem to be a long term solution, there should be update to scripts to handle both scenarios.
